src/wp-includes/global-styles-and-settings.php

```diff
@@ -255,8 +255,46 @@ function wp_add_global_styles_for_blocks() {
 
 	$tree        = WP_Theme_JSON_Resolver::get_merged_data();
 	$block_nodes = $tree->get_styles_block_nodes();
+
+	$can_use_cached = ! wp_is_development_mode( 'theme' );
+	$update_cache   = false;
+
+	if ( $can_use_cached ) {
+		// Hash the merged WP_Theme_JSON data to bust cache on settings or styles change.
+		$cache_hash = md5( wp_json_encode( $tree->get_raw_data() ) );
+		$cache_key  = 'wp_styles_for_blocks';
+		$cached     = get_transient( $cache_key );
+
+		// Reset the cached data if there is no value or if the hash has changed.
+		if ( ! is_array( $cached ) || $cached['hash'] !== $cache_hash ) {
+			$cached = array(
+				'hash'   => $cache_hash,
+				'blocks' => array(),
+			);
+
+			// Update the cache if the hash has changed.
+			$update_cache = true;
+		}
+	}
+
 	foreach ( $block_nodes as $metadata ) {
-		$block_css = $tree->get_styles_for_block( $metadata );
+
+		if ( $can_use_cached ) {
+			// Use the block name as the key for cached CSS data. Otherwise, use a hash of the metadata.
+			$cache_node_key = isset( $metadata['name'] ) ? $metadata['name'] : md5( wp_json_encode( $metadata ) );
+
+			if ( isset( $cached['blocks'][ $cache_node_key ] ) ) {
+				$block_css = $cached['blocks'][ $cache_node_key ];
+			} else {
+				$block_css                           = $tree->get_styles_for_block( $metadata );
```

Comment:
Just to make sure, can the meta data include block settings in the editor that will effect the output styles? 

If so, then this may have incorrect data when a block is updated. If not then it's dandy.

Reply:
I'm not sure I fully grasp the question, sorry.

Global styles for blocks set in the editor are saved to the Global Styles CPT, which are retrieved and merged with base and theme data via `WP_Theme_JSON_Resolver::get_merged_data();` above. While in the editor, if global styles for a block type are adjusted without saving, there's a JS hook that outputs updated styles overriding those enqueued here. 

My understanding is that the styles generated here only need to reflect what is saved for Global Styles.

Not sure if that helps, answers the question, or just confuses things 😅

Reply:
> My understanding is that the styles generated here only need to reflect what is saved for Global Styles.

That helps and was what I was asking, thanks Aaron.